pkg/combustion/rpm.go

```diff
@@ -97,16 +97,48 @@ func configureRPMs(ctx *image.Context) ([]string, error) {
 	return []string{script}, nil
 }
 
-// determine whether RPM configuration is needed
+// SkipRPMComponent determines whether RPM configuration is needed
 func SkipRPMComponent(ctx *image.Context) bool {
 	pkg := ctx.ImageDefinition.OperatingSystem.Packages
 
 	if isComponentConfigured(ctx, userRPMsDir) {
-		// User provided standalone or third party RPMs
+		// isComponentConfigured will indicate if the directory exists, but not
+		// if there are RPMs in there. If there aren't any, it is still possible to
+		// continue if there have been packages specified in the definition.
+		rpmsDir := RPMsPath(ctx)
+
+		dirListing, err := os.ReadDir(rpmsDir)
+		if err != nil {
+			zap.S().Errorf("checking for side-loaded RPMs: %s", err)
```

Comment:
I don't love this, but it's a rare possibility and will at least be logged if we're trying to figure out why the RPM section is skipped.